src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -571,9 +598,27 @@ protected internal void OnAnalyzerConfigDocumentOpened(DocumentId documentId, So
             this.RegisterText(textContainer);
         }
 
+        /// <summary>
+        /// Tries to close the document identified by <paramref name="documentId"/>.  This is only needed by
+        /// implementations of ILspWorkspace to indicate that the workspace should try to transition to the closed state
+        /// for this document, but can bail out gracefully if they don't know about it (for example if they haven't
+        /// heard about the file from the project system).  Subclasses should determine what file contents they should
+        /// transition to if the file is within the workspace.
+        /// </summary>
+        /// <param name="documentId"></param>
+        internal virtual void TryOnDocumentClosed(DocumentId documentId)
+        {
+            throw new NotImplementedException();
+        }
+
 #pragma warning disable IDE0060 // Remove unused parameter 'updateActiveContext' - shipped public API.
         protected internal void OnDocumentClosed(DocumentId documentId, TextLoader reloader, bool updateActiveContext = false)
 #pragma warning restore IDE0060 // Remove unused parameter
+        {
+            OnDocumentClosedEx(documentId, reloader, requireDocumentPresentAndOpen: true);
+        }
+
+        private protected void OnDocumentClosedEx(DocumentId documentId, TextLoader reloader, bool requireDocumentPresentAndOpen)
```

Comment:
Ex as part of the name as otherwise this method has hte same signature as the method above.